### PR TITLE
refactor(app): reimplement protocol run analytics

### LINKED ISSUE #10749: 6.0 Analytics: Hook back up the events related to running protocols
# Overview
A number of Mixpanel events stopped firing after the shift from 4.7.0 to 5.0.x. Let's hook them back up to the new UI  elements.

# Acceptance criteria
Make sure the following events are hooked up to the buttons on the protocol run details page: 
- [x] runStart

- [x] runFinish

- [x] runPause

- [x] runResume

- [x] runCancel

- [x] runAgain (from Protocol Run Detail Page CTA button)

- [x] runAgain (from Protocol Run Detail Page overflow menu)

### PR DESCRIPTION
**Pull request by vabruzzo**
# Overview
Reimplements protocol run event tracking.
Closes #10749

# Changelog
- Adds `useProtocolRunAnalyticsData` hook based on prior deprecated `redux/analytics` actions
- Adds `useRobotAnalyticsData` hook based on prior deprecated `redux/analytics` actions
- Adds `useTrackProtocolRunEvent`
- Adds tracking events for `runStart`, `runPause`, `runResume`, `runCancel`, `runAgain`, and `runFinish`

# Review requests
- ensure events are fired at appropriate times with appropriate data
- refer to `app/src/redux/analytics/make-event.ts` for comments about properties we sent along with each event in the past

# Risk assessment
Low

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/ProtocolRunHeader.tsx [hunk 1]
```diff
@@ -237,7 +253,44 @@ export function ProtocolRunHeader({
       setShowIsShakingModal(true)
     } else if (isHeaterShakerInProtocol && !isShaking) {
       confirmAttachment()
-    } else play()
+    } else {
+      play()
+
+      const isIdle = runStatus === RUN_STATUS_IDLE
+      const eventProperties = isIdle ? { ...robotAnalyticsData } : {}
```
Comment (b-cooper) [inline app/src/organisms/Devices/ProtocolRun/ProtocolRunHeader.tsx]: this spread is unnecessary {...robotAnalyticsData} === robotAnalyticsData

File: app/src/pages/AppSettings/GeneralSettings.tsx [hunk 2]
```diff
@@ -189,6 +189,7 @@ export function GeneralSettings(): JSX.Element {
           <Box>
             <Link
               role="button"
+              tabIndex={0}
```
Comment (b-cooper) [inline app/src/pages/AppSettings/GeneralSettings.tsx]: are we sure that this should have the explicit tabIndex of 0 on the page whenever it is rendered?

**Review by b-cooper — APPROVED** [review]
The changes look good. Thanks for adding these analytics back in 🎉 .

  I like the approach of creating a helper hook to capture the shared behavior of tracking certain run lifecycle concerns.  It looks to me like the promise handling could be more generic so that the hook definition handles errors uniformly. Then each call to `trackProtocolRunEvent` can basically be a 1-liner that passes the event name.  Also properties could just default to `{}` so that we don't have to pass it down for events that down include properties.  These sorts of refactors could definitely be addressed in follow-up PR's.

File: app/src/pages/AppSettings/GeneralSettings.tsx (hunk 2 above)
Comment (vabruzzo) [reply app/src/pages/AppSettings/GeneralSettings.tsx]: I'm removing this, it was something i was testing when i happened to be on this branch, thanks for catching. we did have a discussion about this on slack, a bunch of our links were not accessible via keyboard navigation and koji addressed it in a pr.

**Review by koji — APPROVED** [review]
Looks good to me.
Thank you for adding `JSDoc`. Hopefully, we can add JSDoc to the rest of the hooks at some point.

**Comment by vabruzzo** [comment]
> Looks good to me. Thank you for adding `JSDoc`. Hopefully, we can add JSDoc to the rest of the hooks at some point.

yeah definitely, im hoping to find time in the next week or two to open a pr with jsdocs for all custom hooks, i think this would be good to have and it would be a good way for me to learn about all the hooks i havent run into yet

**Pull request merged**